File: backend-go/internal/models/course_instructor.go
package models

import (
	"time"
)

// CourseInstructor represents the course_instructors table (many-to-many relationship between instructors and course sections)
type CourseInstructor struct {
	ID              uint   `gorm:"primaryKey;autoIncrement" json:"id"`
	InstructorID    uint   `gorm:"column:instructor_id;not null" json:"instructor_id"`
	CourseSectionID uint   `gorm:"column:course_section_id;not null" json:"course_section_id"`
	Role            string `gorm:"default:instructor" json:"role"` // instructor, assistant, coordinator
	CreatedAt       time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt       time.Time `gorm:"autoUpdateTime" json:"updated_at"`

	// Relationships
	Instructor    Instructor    `gorm:"foreignKey:InstructorID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"instructor,omitempty"`
	CourseSection CourseSection `gorm:"foreignKey:CourseSectionID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE" json:"course_section,omitempty"`
}

// TableName specifies the table name for CourseInstructor model
func (CourseInstructor) TableName() string {
	return "course_instructors"
}

File: apps/backend/app/models/course_instructor.ts
import { DateTime } from 'luxon'
import { BaseModel, column, belongsTo } from '@adonisjs/lucid/orm'
import type { BelongsTo } from '@adonisjs/lucid/types/relations'
import Instructor from './instructor.js'
import CourseSection from './course_section.js'

export default class CourseInstructor extends BaseModel {
  public static table = 'course_instructors'

  @column({ isPrimary: true })
  declare id: number

  @column()
  declare course_section_id: number

  @column()
  declare instructor_id: number

  @belongsTo(() => CourseSection, {
    foreignKey: 'course_section_id',
  })
  declare course_section: BelongsTo<typeof CourseSection>

  @belongsTo(() => Instructor, {
    foreignKey: 'instructor_id',
  })
  declare instructor: BelongsTo<typeof Instructor>

  @column.dateTime({ autoCreate: true })
  declare createdAt: DateTime

  @column.dateTime({ autoCreate: true, autoUpdate: true })
  declare updatedAt: DateTime
}
